x/bonds/internal/types/errors.go

```diff
@@ -1,263 +1,53 @@
+	ErrArgumentMustBeInteger                = sdkerrors.Register(DefaultCodespace, 5, "argument must be an integer value")
+	ErrArgumentMustBeBetween                = sdkerrors.Register(DefaultCodespace, 6, "argument must be between")
+	ErrFunctionParameterMissingOrNonFloat   = sdkerrors.Register(DefaultCodespace, 7, "parameter is missing or is not a float")
+	ErrArgumentMissingOrNonFloat            = sdkerrors.Register(DefaultCodespace, 8, "argument is missing or is not a float")
+	ErrArgumentMissingOrNonUInteger         = sdkerrors.Register(DefaultCodespace, 9, "argument is missing or is not an unsigned integer")
+	ErrArgumentMissingOrNonBoolean          = sdkerrors.Register(DefaultCodespace, 10, "argument is missing or is not true or false")
+	ErrIncorrectNumberOfReserveTokens       = sdkerrors.Register(DefaultCodespace, 11, "incorrect number of reserve tokens")
+	ErrIncorrectNumberOfFunctionParameters  = sdkerrors.Register(DefaultCodespace, 12, "incorrect number of function parameters")
+	ErrBondDoesNotExist                     = sdkerrors.Register(DefaultCodespace, 13, "bond does not exist")
+	ErrBondAlreadyExists                    = sdkerrors.Register(DefaultCodespace, 14, "bond already exists")
+	ErrBondTokenIsTaken                     = sdkerrors.Register(DefaultCodespace, 15, "bond token is taken")
+	ErrBondDoesNotAllowSelling              = sdkerrors.Register(DefaultCodespace, 16, "bond does not allow selling at the moment")
+	ErrDidNotEditAnything                   = sdkerrors.Register(DefaultCodespace, 17, "did not edit any bond field")
+	ErrFromAndToCannotBeTheSameToken        = sdkerrors.Register(DefaultCodespace, 18, "from and to tokens cannot be the same token")
+	ErrDuplicateReserveToken                = sdkerrors.Register(DefaultCodespace, 19, "cannot have duplicate tokens in reserve tokens")
+	ErrInvalidStateForAction                = sdkerrors.Register(DefaultCodespace, 20, "cannot perform that action at the current state")
+	ErrUnrecognizedFunctionType             = sdkerrors.Register(DefaultCodespace, 21, "unrecognized function type")
+	ErrInvalidFunctionParameter             = sdkerrors.Register(DefaultCodespace, 22, "invalid function parameter")
+	ErrFunctionNotAvailableForFunctionType  = sdkerrors.Register(DefaultCodespace, 23, "function is not available for the function type")
+	ErrCannotMakeZeroOutcomePayment         = sdkerrors.Register(DefaultCodespace, 24, "cannot make outcome payment because outcome payment is set to nil")
+	ErrFunctionRequiresNonZeroCurrentSupply = sdkerrors.Register(DefaultCodespace, 25, "function requires the current supply to be non zero")
+	ErrTokenIsNotAValidReserveToken         = sdkerrors.Register(DefaultCodespace, 26, "token is not a valid reserve token")
+	ErrMaxSupplyDenomDoesNotMatchTokenDenom = sdkerrors.Register(DefaultCodespace, 27, "token is not a valid reserve token")
```

Comment:
String should be `"max supply denom does not match token denom"`.

Reply:
Correct